packages/shared/src/utils.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Make a map and return a function for checking if a key
+ * is in that map.
+ */
+export function makeMap(
+  str: string,
+  expectsLowerCase?: boolean
+): (key: string) => boolean {
+  const map: Record<string, boolean> = Object.create(null)
+  const list: Array<string> = str.split(',')
+  for (let i = 0; i < list.length; i++) {
+    map[list[i]] = true
+  }
+  return expectsLowerCase ? val => !!map[val.toLowerCase()] : val => !!map[val]
```

Comment:
I think use in operator is a better choice here.
`return expectsLowerCase ? val => val.toLowerCase() in map : val => val in map`

Reply:
I just ran a bench-marking of `in` vs accessing the map. The `in` way to do this would be `20%` more `slower` compared to direct access.

[JSPref reference](https://jsperf.com/in-vs-access-for-plain-obj/1)

Reply:
You are right.